src/storage/src/hummock/cache.rs

```diff
@@ -583,9 +612,19 @@ impl<K: LruKey, T: LruValue> LruCache<K, T> {
         let mut to_delete = vec![];
         let handle = unsafe {
             let mut shard = self.shards[self.shard(hash)].lock();
+            let pending_request = shard.write_request.remove(&key);
             let ptr = shard.insert(key, hash, charge, value, &mut to_delete);
             debug_assert!(!ptr.is_null());
-            CachableEntry::<K, T> {
+            if let Some(que) = pending_request {
+                for sender in que {
+                    (*ptr).add_ref();
+                    let _ = sender.send(CachableEntry {
+                        cache: self.clone(),
```

Comment:
Can we implement `Clone` for `CachableEntry`?

Reply:
Seems we have implemented it. We can use entry.clone() instead.

Reply:
no. it will lock again.

Reply:
Same as the above comments. If we use AtomicUsize for reference counting, a clone won't involve a lock.